api/validate_fields.py

```diff
@@ -0,0 +1,12 @@
+import flask
+
+from app.__init__ import process_system_profile_spec
+
+
+def validate_fields_in_schema(fields):
+    if fields.get("system_profile"):
+        query_fields = list(fields.get("system_profile").keys())
+        system_profile_schema = process_system_profile_spec()
```

Comment:
Oh, I just noticed that this runs the whole processing function, even though it's already done in the app setup. This should instead be:
```suggestion
from app import system_profile_spec


def validate_fields_in_schema(fields):
    if fields.get("system_profile"):
        query_fields = list(fields.get("system_profile").keys())
        system_profile_schema = system_profile_spec()
```